simba/report.py

```diff
@@ -267,6 +291,268 @@ def generate(schedule, scenario, args):
+    ])
+
+    ax.set_xlabel("Ladetyp")
+    ax.set_ylabel("Umläufe")
+    ax.set_title("Verteilung von Gelegenheitslader, Depotlader")
+
+    ax.yaxis.grid(True)
+    plt.savefig(extended_plots_path / "charge_type_proportion")
+    plt.close()
+
+
+def gc_power_time_overview_example(args, scenario):
+
+    gc_list = list(scenario.constants.grid_connectors.keys())
+    for gc in gc_list:
+        # data
+        ts = getattr(scenario, f"{gc}_timeseries")
+        time = ts["time"]
+        total = ts["grid power [kW]"]
+        feed_in = ts["feed-in [kW]"]
+        ext_load = ts["ext.load [kW]"]
+        cs = ts["sum CS power"]
+
+        # plot
+        plt.plot(time, total, label="total")
+        plt.plot(time, feed_in, label="feed_in")
+        plt.plot(time, ext_load, label="ext_load")
+        plt.plot(time, cs, label="CS")
+        plt.legend()
+        plt.xticks(rotation=45)
+
+        plt.savefig(args.results_directory / f"{gc}_power_time_overview")
+        plt.clf()
+
+
+def gc_power_time_overview(extended_plots_path, scenario):
+    """Plots the different loads (total, feedin, external) of all grid connectors.
+
+    :param extended_plots_path: Path argument which combines args.results_directory
+                                and string 'extended_plots'
+    :type extended_plots_path: PosixPath
+    :param scenario: Provides the data for the grid connectors over time.
+    :type scenario: spice_ev.Scenario
+    """
+    gc_list = list(scenario.components.grid_connectors.keys())
+
+    for gc in gc_list:
+        fig, ax = plt.subplots()
+
+        agg_ts = aggregate_timeseries(scenario, gc)
+        headers = [
+            "grid supply [kW]",
+            "fixed load [kW]",
+            "local generation [kW]",
+            "sum CS power [kW]",
+            "battery power [kW]",
+            "bat. stored energy [kWh]",
+        ]
+
+        time_index = agg_ts["header"].index("time")
+        time_values = [row[time_index] for row in agg_ts["timeseries"]]
+
+        for index, header in enumerate(headers):
+            try:
+                header_index = agg_ts["header"].index(header)
+                header_values = [row[header_index] for row in agg_ts["timeseries"]]
+
+                if header == "bat. stored energy [kWh]":
+                    ax2 = ax.twinx()
+                    ax2.set_ylabel("Power in kWh")
+                    next_color = plt.rcParams['axes.prop_cycle'].by_key()["color"][index]
+                    ax2.plot(time_values, header_values, label=header, c=next_color,
+                             linestyle="dashdot")
+                    ax2.legend()
+                    ax2.tick_params(axis='x', rotation=30)
+                    fig.set_size_inches(8, 4.8)
+                else:
+                    ax.plot(time_values, header_values, label=header)
+            except ValueError:
+                continue
+        try:
+            ax2.set_ylim(ax.get_ylim())
+        except NameError:
+            continue
```

Comment:
Why is this needed at all? As I can see, this is only relevant if battery energy is plotted. It does not make sense to have the same y axis for power and energy values.
For future projects, consider initializing ax2 with None or (if there are multiple axes) putting them into an array and then looping over said array.